### ISSUE TITLE
The model and loaded state dict do not match exactly, but the config I use is correct

### ISSUE THREAD

**Issue by yongquan-huang** [open-mmlab/mmdetection#7552]
**Describe the bug**
when i run `test.py`, the model and loaded state dict do not match exactly, but the config I use is correct
and there are test results:

- load checkpoint from local path: work_dirs/yolox_x_1280_2b_300e_teaBudBlight/best_bbox_mAP_epoch_170.pth
The model and loaded state dict do not match exactly
unexpected key in source state_dict: ema_backbone_stem_conv_conv_weight, ema_backbone_stem_conv_bn_weight, ema_backbone_stem_conv_bn_bias, ema_backbone_stem_conv_bn_running_mean, ema_backbone_stem_conv_bn_running_var, 
......
ema_bbox_head_multi_level_conv_obj_2_bias
[                                                  ] 0/76, elapsed: 0s, ETA:/home/zc/pythonProject/hyq/venv/lib/python3.8/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 76/76, 0.9 task/s, elapsed: 81s, ETA:     0s
Evaluating bbox...
Loading and preparing results...
DONE (t=0.01s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.06s).
Accumulating evaluation results...
DONE (t=0.04s).
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
......
OrderedDict([('bbox_mAP', 0.001), ('bbox_mAP_50', 0.002), ('bbox_mAP_75', 0.0), ('bbox_mAP_s', -1.0), ('bbox_mAP_m', 0.0), ('bbox_mAP_l', 0.001), ('bbox_mAP_copypaste', '0.001 0.002 0.000 -1.000 0.000 0.001')])

**Reproduction**

1. run `test.py` with parameters

```
/home/zc/pythonProject/hyq/venv/bin/python /home/zc/pythonProject/hyq/mmdetection/tools/test.py mmdetection/configs/yolox/yolox_x_1280_2b_300e_teaBudBlight.py work_dirs/yolox_x_1280_2b_300e_teaBudBlight/best_bbox_mAP_epoch_170.pth --show-dir work_dirs/yolox_x_1280_2b_300e_teaBudBlight/val_results --eval bbox
```

2. run `train.py` with paramters
```
/home/zc/pythonProject/hyq/venv/bin/python /home/zc/pythonProject/hyq/mmdetection/tools/train.py mmdetection/configs/yolox/yolox_x_1280_2b_300e_teaBudBlight.py --work-dir work_dirs/yolox_x_1280_2b_300e_teaBudBlight
```

3. config
my config is based on the `yolox_s_8x8_300e_coco.py` , modify into `yolox_x`, and then modify some required parameters when i train my dataset

```
_base_ = ['../_base_/schedules/schedule_1x.py',
          '../_base_/default_runtime.py']
img_scale = (1280, 1280)  # 640，640
 # model settings
model = dict(
    type='YOLOX',
    input_size=img_scale,
    random_size_range=(15, 25),
    random_size_interval=10,
    backbone=dict(type='CSPDarknet', deepen_factor=1.33, widen_factor=1.25),
    neck=dict(
        type='YOLOXPAFPN',
        in_channels=[320, 640, 1280],
        out_channels=320,
        num_csp_blocks=4),
    bbox_head=dict(
        type='YOLOXHead', num_classes=2, in_channels=320, feat_channels=320
    ),
    train_cfg=dict(assigner=dict(type='SimOTAAssigner', center_radius=2.5)),
    # In order to align the source code, the threshold of the val phase is
    # 0.01, and the threshold of the test phase is 0.001.
    test_cfg=dict(score_thr=0.01, nms=dict(type='nms', iou_threshold=0.65)))
# dataset settings
data_root = 'mmdetection/data/'
......
```

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
```
sys.platform: linux
Python: 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
CUDA available: True
GPU 0: NVIDIA GeForce RTX 3090
CUDA_HOME: /usr/local/cuda-10.0
NVCC: Cuda compilation tools, release 10.0, V10.0.130
GCC: gcc (Ubuntu 7.5.0-6ubuntu2) 7.5.0
PyTorch: 1.9.0+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.10.0+cu111
OpenCV: 4.5.5
MMCV: 1.3.17
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.1
MMDetection: 2.22.0+70f6d9c
```

**Comment by Czm369** [comment]
The ema model is necessary for resuming, so you can delete the ema model when running `test.py`.

**Comment by yongquan-huang** [comment]
@Czm369 
sorry，how to delete the ema model？
and i see the `The saved Model parameter is actually the ema Model parameter value, and the EMA Model parameter is the model parameter value`，Does this mean that the weights in my model above were successfully loaded？

**Comment by RangiLyu** [comment]
> @Czm369 sorry，how to delete the ema model？ and i see the `The saved Model parameter is actually the ema Model parameter value, and the EMA Model parameter is the model parameter value`，Does this mean that the weights in my model above were successfully loaded？

No need to worry about the warning. The ema buffers are not used during testing.

**Comment by yongquan-huang** [comment]
> > @Czm369 sorry，how to delete the ema model？ and i see the `The saved Model parameter is actually the ema Model parameter value, and the EMA Model parameter is the model parameter value`，Does this mean that the weights in my model above were successfully loaded？
> 
> No need to worry about the warning. The ema buffers are not used during testing.

thanks
